### PR TITLE
docs: Add Terraform tutorial

### PR DESCRIPTION
**Pull request by timothyhinrichs**

### CODE REVIEW

File: docs/book/terraform.md [hunk 1]
```diff
@@ -0,0 +1,302 @@
+# Terraform Testing
+
+Terraform lets you describe the infrastructure you want and automatically creates, deletes, and modifies
+your existing infrastructure to match.  OPA makes it possible to write policies that test the changes
+Terraform is about to make before it makes them.  Such tests help in different ways:
+* tests help individual developers sanity check their terraform changes
+* tests can auto-approve run-of-the-mill infrastructure changes and reduce
+the burden of peer-review
+* tests can help catch problems that arise when applying terraform to production after applying it to staging
```
Comment (tsandall) [inline docs/book/terraform.md]: Nit: use consistent case for Terraform throughout.

File: docs/book/terraform.md [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+
+Create a [Terraform](https://www.terraform.io/docs/index.html) file that includes an 
+auto-scaling group and a server on AWS.  (You will need to modify the `shared_credentials_file`
+to point to your AWS credentials.)
+
+```shell
+cat >main.tf <<EOF
+provider "aws" {
+    region = "us-west-1"
+    shared_credentials_file = "/Users/alice/.aws/credentials"
```
Comment (tsandall) [inline docs/book/terraform.md]: Nit: replace `/Users/alice` with `$HOME/.aws/credentials`.

File: docs/book/terraform.md [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+EOF
+```
+
+### 4. Evaluate the OPA policy on the Terraform plan
+
+To evaluate the policy against that plan, you hand OPA the policy, the Terraform plan as input, and
+ask it to evaluate `data.terraform.analysis.authz`.
+
+```shell
+opa run terraform.rego repl.input:tfplan.json -e "data.terraform.analysis.authz"
```
Comment (tsandall) [inline docs/book/terraform.md]: Can we generate another plan that fails the authorization check?

File: docs/book/terraform.md (hunk 2 above)
Comment (hauleth) [inline docs/book/terraform.md]: This isn't needed at all as Terraform knows where to find it anyway.

Comment (tsandall) [inline docs/book/terraform.md]: Either way works for me as long as `/Users/alice` is removed.

**Comment by timothyhinrichs** [comment]
Comments addressed in new commit.

**Comment by tsandall** [comment]
LGTM

**Pull request merged**